Fix deserialization of nullable scalar fields (#16)
Crell/Serde: README.md

PHP objects can be mutated to and from a serialized format.  Nested objects can be flattened or collected, classes with common interfaces can be mapped to the appropriate object, and array values can be imploded into a string for serialization and exploded back into an array when reading.

> Null values are supported in both directions, but beware of the difference between `null` and uninitialized (i.e. not provided field), ensure proper default values or proper usage.

Reviewer: This sentence doesn't parse for me.

Author: Second attempt

In this case, the `$other` property has two keys, `foo` and `bar`, with values `beep` and `boop`, respectively.  The same JSON will deserialize back to the same object as before.

> In case of serialization of property with `flatten` set to true but having `null` value, no property will be serialized. Yet on deserialization the object of uninitialized properties will be created.

Reviewer: This sentence doesn't parse for me.

Author: Rephrased and added exhausting example.

In this case, the `$other` property has two keys, `foo` and `bar`, with values `beep` and `boop`, respectively.  The same JSON will deserialize back to the same object as before.

With flattened object the serialization and deserialization is asymmetrical, consider this example:

> ```php

Reviewer: ```suggestion

```php
```

class Page
{
    public function __construct(
        public int $total,
        public int $limit,
        #[Serde\Field(flatten: true)]
        public null|PaginationState $state,
    ) {}
}

class PaginationState
{
    public function __construct(
        public int $offset,
    ) {
    }
}

> $page = new Page(100, 10, null);

Reviewer: ```suggestion

$page = new Page(100, 10, null);
```

```

> Will serialized with omitted the flattened object into this JSON payload

Reviewer: ```suggestion
This object will serialize into the following JSON:
```

```json
{
    "total":100,
    "limit":10
}
```

> However, this payload will be deserialized intot the following object, especially notice the uninitialized DTO:

Reviewer: ```suggestion
However, when deserializing back into PHP, the `$state` property will get a new, uninitialized `PaginationState` instance created.  If you have a nullable property, be aware that it may end up with an uninitialized object.
```

Reviewer: That said... I really don't like this behavior.  It seems like a foot gun waiting to happen.
